src/app/components/Tokens/TokenHolders.tsx

```diff
@@ -0,0 +1,90 @@
+import { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Table, TableCellAlign, TableColProps } from '../Table'
+import { BareTokenHolder } from '../../../oasis-nexus/api'
+import { TablePaginationProps } from '../Table/TablePagination'
+import { AccountLink } from '../Account/AccountLink'
+import { fromBaseUnits } from '../../utils/helpers'
+import { RoundedBalance } from '../RoundedBalance'
+
+type TableTokenHolder = BareTokenHolder & {
+  markAsNew?: boolean
+}
+
+type TokenHoldersProps = {
+  holders?: TableTokenHolder[]
+  isLoading: boolean
+  decimals: number
+  totalSupply: string | undefined
+  limit: number
+  pagination: false | TablePaginationProps
+}
+
+export const TokenHolders: FC<TokenHoldersProps> = ({
+  isLoading,
+  limit,
+  pagination,
+  holders,
+  decimals,
+  totalSupply,
+}) => {
+  const { t } = useTranslation()
+  const tableColumns: TableColProps[] = [
+    { key: 'rank', content: t('common.rank'), align: TableCellAlign.Center },
+    { key: 'address', content: t('common.address') },
+    { key: 'quantity', content: t('common.quantity'), align: TableCellAlign.Right },
+    { key: 'percentage', content: t('common.percentage'), align: TableCellAlign.Right },
+  ]
+
+  const calculateRatio = (balance: string): string => {
+    return totalSupply === undefined
+      ? t('common.missing')
+      : `${((100 * parseFloat(fromBaseUnits(balance, decimals))) / parseFloat(totalSupply)).toFixed(4)}%`
+  }
+
+  const tableRows = holders?.map((holder, index) => {
+    return {
+      key: holder.holder_address,
+      data: [
+        {
+          key: 'rank',
+          content: (
+            index +
+            (pagination ? (pagination.selectedPage - 1) * pagination.rowsPerPage : 0) +
+            1
+          ).toLocaleString(),
+          align: TableCellAlign.Center,
+        },
+        {
+          key: 'address',
+          content: (
+            <AccountLink scope={holder} address={holder.eth_holder_address || holder.holder_address} />
+          ),
+        },
+        {
+          key: 'quantity',
+          content: <RoundedBalance value={fromBaseUnits(holder.balance, decimals)} />,
```

Comment:
Well it's problematic, because in api.is, when parsing the results of the get token holders call, we don't have access to the definition of the token, which defines the number of decimals. _Maybe_ we could make it into a required argument of the wrapper function call, but then it would change the signature, which is not nice, and also semantically doesn't make all that much sense.

Reply:
@lukaw3d , would you prefer if I added decimals as a required parameter to this call, so that I can use it to normalize the data in the API layer?

Reply:
no, rather keep it as is